xpra/gtk/clipboard.py

```diff
@@ -155,8 +155,14 @@ def get_contents(self, target: str, got_contents: ClipboardCallback, time=0) ->
                 got_contents(target, 8, text)
                 return
         else:
-            # data = wait_for_contents(target)?
-            pass
+            r = self.clipboard.wait_for_targets()
+            if r and len(r) == 2 and r[0]:
+                atom = next((x for x in r[1] if x.name() == target), None)
+                if atom != None:
+                    sel = self.clipboard.wait_for_contents(atom)
```

Comment:
According to the docs, this function can return `None`:
https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Clipboard.html#Gtk.Clipboard.wait_for_contents